File: src/dataset/typescript/armstrong-numbers_Exercism-typescript-shybyte/gemini/GeminiFlash_armstrong-numbers_v4.ts
function sum(numbers: number[]) {
    let total = 0;
    for (let i = 0; i < numbers.length; i++) {
        total += numbers[i];
    }
    return total;
}

export default class ArmstrongNumbers {
    public static isArmstrongNumber(x: number) {
        const numStr = x.toString();
        const numDigits = numStr.length;
        let armstrongSum = 0;

        for (let i = 0; i < numDigits; i++) {
            const digit = parseInt(numStr[i], 10);
            armstrongSum += digit ** numDigits;
        }

        return x === armstrongSum;
    }
}

File: src/dataset/c/armstrong-numbers_exercism-alanmanderson/gemini/GeminiFlash_armstrong_numbers_v4.c
#include "armstrong_numbers.h"
#include <math.h>
#include <stdbool.h>

bool is_armstrong_number(int candidate) {
  if (candidate < 0) {
    return false;
  }

  int num_digits = 0;
  int temp = candidate;
  while (temp != 0) {
    temp /= 10;
    num_digits++;
  }

  int sum = 0;
  temp = candidate;
  while (temp != 0) {
    int digit = temp % 10;
    sum += round(pow(digit, num_digits));
    temp /= 10;
  }

  return sum == candidate;
}